app/plugin.go

```diff
@@ -210,9 +210,13 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	// Sync plugin active state when config changes. Also notify plugins.
 	a.Srv().PluginsLock.Lock()
 	a.RemoveConfigListener(a.Srv().PluginConfigListenerId)
-	a.Srv().PluginConfigListenerId = a.AddConfigListener(func(*model.Config, *model.Config) {
-		a.installFeatureFlagPlugins()
-		a.SyncPluginsActiveState()
+	a.Srv().PluginConfigListenerId = a.AddConfigListener(func(old, new *model.Config) {
+		// If plugin status remains unchanged, only then run this.
+		// Because (*App).InitPlugins is already run as a config change hook.
+		if *old.PluginSettings.Enable == *new.PluginSettings.Enable {
```

Comment:
Is this gate still relevant? Wondering how individual plugins being enabled or disabled actually happen without the `a.SyncPluginsActiveState`? (Or is that in the "other" config listener?)

Reply:
Yes, this is relevant to prevent the same code from being run twice.

> Wondering how individual plugins being enabled or disabled actually happen without the a.SyncPluginsActiveState? (Or is that in the "other" config listener?)

Yes. See my PR message :wink: 

> And that
calls a.SyncPluginsActiveState(). But inside a.InitPlugins,
there is yet another config listener which again calls
a.SyncPluginsActiveState(). The first fix is to simply not call
the method again from the nested listener.